handler/parse.go

```diff
@@ -36,74 +37,89 @@ func parseData(r *http.Request) (dataTree, error) {
 }
 
 // pushes value into data tree.
-func (d dataTree) push(k string, v []string) error {
+func (dt dataTree) push(k string, v []string) error {
 	keys := fetchIndexes(k)
 	if len(keys) <= MaxLevel {
-		return d.mount(keys, v)
+		return dt.mount(keys, v)
 	}
 	return nil
 }
 
-func prepareNewDataNode(d dataTree, i []string, v []string) (bool, error) {
-	p, ok := d[i[0]]
-	_, dataTreeOK := p.(dataTree)
-	if ok && dataTreeOK && len(i) == 1 && (len(v) == 0 || len(v[0]) == 0) {
-		return false, nil
+func prepareNewDataNode(dt dataTree, i []string, v []string) (bool, error) {
+	if len(i) == 0 {
+		return false, errors.New("should not prepare new node")
 	}
-	if ok && !dataTreeOK {
-		oldV := d[i[0]]
-		oldVString, stringOk := oldV.(string)
-		oldVStringArray, stringArrayOk := oldV.([]string)
 
-		switch {
-		case stringOk && len(oldVString) == 0,
-			stringArrayOk && len(oldVStringArray) == 0:
-			d[i[0]] = make(dataTree)
-		default:
-			return false, fmt.Errorf(
-				"invalid value in dataTree. key: %+v, val: %+v, tree: %+v",
-				i[0],
-				v,
-				d,
-			)
+	_, ok := dt[i[0]]
+	if !ok {
+		dt[i[0]] = make(dataTree)
+		return true, nil
+	}
+
+	potentialErr := fmt.Errorf(
```

Comment:
Do not preallocate an error. This is an `unlikely` branch, so you'll usually end up with the costly `fmt` allocation for no reason 😃.

Reply:
Got it